File: 200_LeetCode_Questions/Java/D11P2_LeetCode_153.java
# Java code coming soon
class Solution {
    public int findMin(int[] nums) {
        int l = 0, r = nums.length-1;
        while(l < r){
            int mid = l + (r-l)/2;
            if(nums[mid] > nums[r]) l = mid+1;
            else if(nums[mid] < nums[r]) r = mid;
            else r--;
        }
        return nums[l];
    }
}

File: 200_LeetCode_Questions/Python/D11P2_LeetCode_153.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        left = 0
        right = len(nums) - 1

        while (left < right):
            midpoint = int(left + (right - left)/2)
            if(nums[midpoint] < nums[midpoint -1]):
                return nums[midpoint]
            elif (nums[midpoint] >= nums[left] and nums[midpoint] > nums[right]):
                left = midpoint+1
            else:
                right = midpoint - 1

        return nums[left]
